src/lib/common/sem.cpp

```diff
@@ -459,13 +459,131 @@ int timeStatSemGive(const char* who, const char* what)
 
 
 /* ****************************************************************************
+*
+* reqSemGet - 
+*/
+const char* reqSemGet(void)
+{
+  int value;
+
+  if (sem_getvalue(&reqSem, &value) == -1)
+  {
+    return "error - state not available";
+  }
+
+  if (value == 0)
+  {
+    return "taken";
+  }
+  
+  return "free";  
+}
+
+
+
+/* ****************************************************************************
+*
+* transSemGet - 
+*/
+const char* transSemGet(void)
+{
+  int value;
+
+  if (sem_getvalue(&transSem, &value) == -1)
+  {
+    return "error - state not available";
+  }
+
+  if (value == 0)
+  {
+    return "taken";
+  }
+  
+  return "free";  
+}
+
+
+
+/* ****************************************************************************
+*
+* cacheSemGet - 
+*/
+const char* cacheSemGet(void)
+{
+  int value;
+
+  if (sem_getvalue(&cacheSem, &value) == -1)
+  {
+    return "error - state not available";
+  }
+
+  if (value == 0)
+  {
+    return "taken";
+  }
+  
+  return "free";  
+}
+
+
+
+/* ****************************************************************************
+*
+* timeStatSemGet - 
+*/
+const char* timeStatSemGet(void)
+{
+  int value;
+
+  if (sem_getvalue(&timeStatSem, &value) == -1)
+  {
+    return "error - state not available";
+  }
+
+  if (value == 0)
+  {
+    return "taken";
+  }
+  
+  return "free";  
+}
+
+
+
+/* ****************************************************************************
 *  curl context
 */
 
+static bool contexts_mutex_taken   = false;
```

Comment:
Yeah, you know how much I hate the underscores ...
However, you have to look at 'the whole picture' here, there are already 'closely related' variables with underscores. These vars should have really similar names, otherwise it gets a bit confusing.
So, either I change name of all of them (5 vars in total), or I leave it as is